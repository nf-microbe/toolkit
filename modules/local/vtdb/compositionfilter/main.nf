process VTDB_COMPOSITIONFILTER {
    tag "${meta.id}"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/mulled-v2-80c23cbcd32e2891421c54d1899665046feb07ef:77a31e289d22068839533bf21f8c4248ad274b60-0' :
        'biocontainers/mulled-v2-80c23cbcd32e2891421c54d1899665046feb07ef:77a31e289d22068839533bf21f8c4248ad274b60-0' }"

    input:
    tuple val(meta), path(combined_data)
    path composition_filters

    output:
    tuple val(meta), path("${prefix}_composition_data.tsv") , emit: compos_data
    path "versions.yml"                                     , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    compositionfilter.py \\
        --input ${combined_data} \\
        --filters ${composition_filters} \\
        --output ${prefix}_composition_data.tsv \\
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version | sed 's/Python //' )
        pandas: \$(echo \$(pandas_version.py 2>&1))
        biopython: \$(echo \$(biopython_version.py 2>&1))
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_composition_data.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version | sed 's/Python //' )
        pandas: \$(echo \$(pandas_version.py 2>&1))
        biopython: \$(echo \$(biopython_version.py 2>&1))
    END_VERSIONS
    """
}
