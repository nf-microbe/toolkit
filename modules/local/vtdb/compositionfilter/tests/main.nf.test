nextflow_process {

    name "Test Process: COMPOSITIONFILTER"
    script "../main.nf"
    process "COMPOSITIONFILTER"

    test("combined.tsv + composition_filters.tsv") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/compositionfilter/test_class_data.tsv", checkIfExists: true)
                ]
                input[1] = file("${projectDir}/assets/filters/composition_filters.tsv", checkIfExists: true)
                """
            }
        }
        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ],
                    file("${projectDir}/assets/test-datasets/modules/local/compositionfilter/test_class_data.tsv", checkIfExists: true)
                ]
                input[1] = file("${projectDir}/assets/filters/composition_filters.tsv", checkIfExists: true)
                """
            }
        }
        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
