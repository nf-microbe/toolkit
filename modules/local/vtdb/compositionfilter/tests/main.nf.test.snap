{
    "combined.tsv + composition_filters.tsv": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_composition_data.tsv:md5,805029c01a31b89682805b70abfc9611"
                    ]
                ],
                "1": [
                    "versions.yml:md5,32360aa25116c8df5029482128c3919a"
                ],
                "compos_data": [
                    [
                        {
                            "id": "test"
                        },
                        "test_composition_data.tsv:md5,805029c01a31b89682805b70abfc9611"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,32360aa25116c8df5029482128c3919a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-14T16:43:44.373739127"
    },
    "stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_composition_data.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,32360aa25116c8df5029482128c3919a"
                ],
                "compos_data": [
                    [
                        {
                            "id": "test"
                        },
                        "test_composition_data.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,32360aa25116c8df5029482128c3919a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-15T07:26:50.554381172"
    }
}